apps/studio/src/lib/db/clients/bigquery.js

```diff
@@ -181,6 +287,34 @@ async function executeQuery(client, queries, job) {
   return results.map(parseRowQueryResult)
 }
 
+async function executeWithTransaction(conn, queryArgs) {
+  let fullQuery = joinQueries([
+    'BEGIN TRANSACTION', queryArgs.query, 'COMMIT TRANSACTION;', `
+      EXCEPTION WHEN ERROR THEN
+      ROLLBACK TRANSACTION;
+      RAISE USING MESSAGE = @@error.message;
+    `,
+    'END',
+    'CALL BQ.ABORT_SESSION()',
+  ]);
+  // if there's a ; after BEGIN, BQ gets really mad
+  fullQuery = 'BEGIN\n' + fullQuery;
```

Comment:
this seems like it will generate a query that looks like 
```sql
BEGIN
BEGIN TRANSACTION
...
```

Is this intended?

Reply:
Yeah this is intended. 

The `BEGIN`  is for the exception block that's in there